namerd/storage/consul/src/main/scala/io/buoyant/namerd/storage/consul/ConsulDtabStore.scala

```diff
@@ -1,106 +1,86 @@
 package io.buoyant.namerd
 package storage.consul
 
-import com.twitter.finagle.{Dtab, Path}
+import com.twitter.finagle.{Dtab, Failure, Path}
 import com.twitter.io.Buf
 import com.twitter.util._
 import io.buoyant.consul.v1._
 import io.buoyant.namerd.DtabStore.{DtabNamespaceAlreadyExistsException, DtabNamespaceDoesNotExistException, DtabVersionMismatchException, Version}
 
-class ConsulDtabStore(api: KvApi, root: Path) extends DtabStore {
+class ConsulDtabStore(api: KvApi, root: Path, datacenter: Option[String] = None) extends DtabStore {
 
-  override def list(): Activity[Set[Ns]] = {
-    def namespace(key: String): Ns = key.stripPrefix("/").stripSuffix("/").substring(root.show.length)
+  def path(ns: Ns) = s"${root.show}/$ns"
 
-    val run = Var.async[Activity.State[Set[Ns]]](Activity.Pending) { updates =>
-      @volatile var running = true
+  def getNs(ns: Ns, index: Option[String] = None) =
```

Comment:
Are you ok with making it `protected`? The reason why I'm asking is one of the reasons for this refactoring - when we had chat with @olix0r and @wmorgan I asked about submitting a PR to make this refactoring and make it easier to extend ConsulDtabStore so we can get our own plugin that would do #555 but without the need to re-write the whole store.